### PR TITLE
Added drive and swapspace tests

### PR DESCRIPTION
**Pull request by japokorn**
added tests for following Dbus api functions:
        * Drive.Eject
        * Drive.PowerOff
        * Drive.SetConfiguration
        * Swapspace.Start
        * Swapspace.Stop

### CODE REVIEW

**Comment by vpodzime** [comment]
Jenkins, this is ok to test.

**Comment by vpodzime** [comment]
Please check the test results with failures like this:
```
FileNotFoundError: [Errno 2] No such file or directory: '/sys/bus/pseudo/drivers/scsi_debug/ptype'
```

**Comment by japokorn** [comment]
FileNotFoundErrors were caused by switching from scsi_debug to targetcli. Fixed.

**Comment by japokorn** [comment]
Jenkins, test this please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

File: src/tests/dbus-tests/test_90_swapspace.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+
+    def get_swapspace_iface(self, device):
+        iface = dbus.Interface(device, dbus_interface=self.iface_prefix + '.Swapspace')
+        return iface
+
+
+    def setUp(self):
+        # create new fake swap device
+        subprocess.call(['modprobe', 'scsi_debug'])
```
Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_90_swapspace.py]: Do you really need scsi_debug devices here? targetcli devices (`self.vdevs`) should work here.

File: src/tests/dbus-tests/test_90_swapspace.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        new_dirs = list(set(new_dirs) - set(dirs))
+        assert len(new_dirs) == 1
+
+        self.dev = os.listdir(new_dirs[0])
+        assert len(self.dev) == 1
+        self.dev = '/dev/' + self.dev[0]
+        assert os.path.exists(self.dev)
+
+        print('\n')
+        subprocess.call(['mkswap', self.dev])
```
Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_90_swapspace.py]: You should create the swap using storaged. Basic test for formatting devices is in `StoragedBlockTest` but it doesn't test creating swaps.

File: src/tests/dbus-tests/test_90_swapspace.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        self.iface.Stop(self.no_options)
+        active = self.get_property(self.iface, '.Swapspace', 'Active')
+        self.assertNotEqual(active, True) # swap shouldn't be active
+
+        self.iface.Start(self.no_options)
+        active = self.get_property(self.iface, '.Swapspace', 'Active')
+        self.assertEqual(active, True) # swap should be active
+
+        # cleanup
+        self.iface.Stop(self.no_options)
```
Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_90_swapspace.py]: Cleanup should be part of `tearDown` (or you can use `addCleanup`) -- if one of the assertion fails the stop method won't be executed. It also might be better to call `swapoff` in cleanup.

File: src/tests/dbus-tests/test_90_swapspace.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            subprocess.call(['udevadm', 'settle'])
+            subprocess.call(['modprobe', '-r', 'scsi_debug'])
+
+
+    def test_10_start_stop(self):
+        # Active property is now in undefined state
+
+        self.iface.Start(self.no_options)
+        active = self.get_property(self.iface, '.Swapspace', 'Active')
+        self.assertEqual(active, True) # swap should be active
```
Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_90_swapspace.py]: You should check "system" values too (not only dbus properties), `swapon` can list all active swap devices.

File: src/tests/dbus-tests/test_40_drive.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+
+    def test_30_setconfiguration(self):
+        ''' Test of Drive.SetConfiguration method '''
+        # set configuration value to some improbable value
+        self.cd_drive.SetConfiguration({'ata-pm-standby': 286}, self.no_options)
+
+        # validate that configration property has changed
+        conf_value = self.get_property(self.cd_drive, '.Drive', 'Configuration')
+        self.assertEqual(str(conf_value['ata-pm-standby']), '286')
```
Comment (vojtechtrefny) [inline src/tests/dbus-tests/test_40_drive.py]: `Drive` interface has a lot of properties, we should test at least some of them (compare dbus properties with some "system" values accessible using `lsblk` etc.)

File: src/tests/dbus-tests/test_40_drive.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+    def get_drive(self, device):
+        drive_name = self.get_drive_name(device)
+        drive = self.get_object('', '/drives/' + os.path.basename(drive_name))
+        drive_object = dbus.Interface(drive, dbus_interface=self.iface_prefix + '.Drive')
+        return drive_object
+
+    def setUp(self):
+        # create new fake CD-ROM
+        subprocess.call(['modprobe', 'scsi_debug'])
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        dirs = glob.glob('/sys/bus/pseudo/drivers/scsi_debug/adapter*/host*/target*/*:*/block')
+
+        with open('/sys/bus/pseudo/drivers/scsi_debug/ptype', 'w') as f:
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+            f.write('5')  # '5' == created devices will be CD drives
+        with open('/sys/bus/pseudo/drivers/scsi_debug/num_tgts', 'w') as f:
+            f.write('1')  # change desired number of new targets to 1
+        with open('/sys/bus/pseudo/drivers/scsi_debug/add_host', 'w') as f:
+            f.write('1')  # generate a new device (add_host also applies changes)
+        subprocess.call(['udevadm', 'settle'])
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+        new_dirs = []
+        # rescan directories; we cannot tell which file is the new one...
+        while len(new_dirs) < len(dirs) + 1:
+            new_dirs = glob.glob('/sys/bus/pseudo/drivers/scsi_debug/adapter*/host*/target*/*:*/block')
+            time.sleep(0.1)
+
+        # ...so we just diff all devices with the previous scan result to get the new one
+        new_dirs = list(set(new_dirs) - set(dirs))
+        assert len(new_dirs) == 1
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+        for d in self.vdevs + [self.cd_dev]:
+            dev = self.get_device(d)
+            drive = self.get_drive(dev)
+            drive_name = self.get_drive_name(dev)
+            if drive_name == self.cd_drive_name:
+                # ejectable devices should support the Eject function
+                try:
+                    drive.Eject(self.no_options)
+                except Exception:
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+        with open('/sys/bus/pseudo/drivers/scsi_debug/ptype', 'w') as f:
+            f.write('5')  # '5' == created devices will be CD drives
+        with open('/sys/bus/pseudo/drivers/scsi_debug/num_tgts', 'w') as f:
+            f.write('1')  # change desired number of new targets to 1
+        with open('/sys/bus/pseudo/drivers/scsi_debug/add_host', 'w') as f:
+            f.write('1')  # generate a new device (add_host also applies changes)
+        subprocess.call(['udevadm', 'settle'])
+
+        new_dirs = []
+        # rescan directories; we cannot tell which file is the new one...
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+            subprocess.call(['modprobe', '-r', 'scsi_debug'])
+
+
+    def test_10_eject(self):
+        ''' Test of Drive.Eject method '''
+        # when new ejectable device is added using scsi_debug,
+        # it affects already existing virtual devices making some of them
+        # ejectable as well; this is a workaround
+
+        for d in self.vdevs + [self.cd_dev]:
```
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

File: src/tests/dbus-tests/test_90_swapspace.py (hunk 1 above)
Comment (japokorn) [inline src/tests/dbus-tests/test_90_swapspace.py]: Fixed.

File: src/tests/dbus-tests/test_90_swapspace.py (hunk 2 above)
Comment (japokorn) [inline src/tests/dbus-tests/test_90_swapspace.py]: Fixed.

File: src/tests/dbus-tests/test_90_swapspace.py (hunk 4 above)
Comment (japokorn) [inline src/tests/dbus-tests/test_90_swapspace.py]: Fixed.

File: src/tests/dbus-tests/test_90_swapspace.py (hunk 3 above)
Comment (japokorn) [inline src/tests/dbus-tests/test_90_swapspace.py]: Fixed.

File: src/tests/dbus-tests/test_40_drive.py (hunk 5 above)
Comment (japokorn) [inline src/tests/dbus-tests/test_40_drive.py]: Fixed.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Pull request merged**